src/app/services/xml-parsers/named-entities-parser.service.ts

```diff
@@ -0,0 +1,343 @@
+import { Injectable } from '@angular/core';
+import { AttributesMap } from 'ng-dynamic-component';
+import { combineLatest } from 'rxjs';
+import { map, shareReplay } from 'rxjs/operators';
+import { AppConfig } from '../../app.config';
+import { GenericElementComponent } from '../../components/generic-element/generic-element.component';
+import {
+  Description, NamedEntitiesList, NamedEntity, NamedEntityInfo,
+  NamedEntityLabel, NamedEntityType, Relation, XMLElement,
+} from '../../models/evt-models';
+import { isNestedInElem, xpath } from '../../utils/dom-utils';
+import { EditionDataService } from '../edition-data.service';
+import { GenericParserService } from './generic-parser.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class NamedEntitiesParserService {
+  public readonly parsedLists$ = this.editionDataService.parsedEditionSource$.pipe(
+    map((source) => this.parseLists(source)),
+    shareReplay(1),
+  );
+
+  public readonly persons$ = this.parsedLists$.pipe(
+    map(({ lists, entities }) => (this.getResultsByType(lists, entities, ['person', 'personGrp']))),
+  );
+
+  public readonly places$ = this.parsedLists$.pipe(
+    map(({ lists, entities }) => this.getResultsByType(lists, entities, ['place'])),
+  );
+
+  public readonly organizations$ = this.parsedLists$.pipe(
+    map(({ lists, entities }) => this.getResultsByType(lists, entities, ['org'])),
+  );
+
+  public readonly relations$ = this.parsedLists$.pipe(
+    map(({ relations }) => relations),
+  );
+
+  public readonly events$ = this.parsedLists$.pipe(
+    map(({ lists, entities }) => this.getResultsByType(lists, entities, ['event'])),
+  );
+
+  public readonly namedEntities$ = combineLatest([
+    this.persons$,
+    this.places$,
+    this.organizations$,
+    this.relations$,
+    this.events$,
+  ]).pipe(
+    map(ne => ({
```

Comment:
`ne` can be `[persons, places, organizations, relations, events]`

Reply:
Do you mean that I should define a more specific type for `ne`?

Reply:
no, you can rewrite the line from
```
map(ne => ...)
```
to
```
map(([persons, places, organizations, relations, events]) => ...)
```